test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/ODataQueryOptionTest.cs

```diff
@@ -264,6 +264,37 @@ public void CanExtractQueryOptionsCorrectly()
             Assert.Equal("DeltaToken", queryOptions.RawValues.DeltaToken);
         }
 
+
+        [Theory]
+        [InlineData(" $filter=Filter& $select=Select& $orderby=OrderBy& $expand=Expand& $top=10& $skip=20& $count=true& $skiptoken=SkipToken& $deltatoken=DeltaToken")]
```

Comment:
>$filter=Filter& $select=Select& $orderby=OrderBy [](start = 22, length = 48)

Maybe add the example of tailing ?

Reply:
Last InlineData  (line 271)  does trailing `$filter =Filter...`

Reply:
I added two more cases trailing %20  and spaces before and after